src/main/java/com/patchworkmc/mapping/IntermediaryHolder.java

```diff
@@ -0,0 +1,71 @@
+package com.patchworkmc.mapping;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.fabricmc.tinyremapper.IMappingProvider;
+
+/**
+ * A lazily loading representation of all Minecraft names and descriptors.
+ */
+public class IntermediaryHolder {
+	private final HashMap<String, Map<String, Member>> mappings = new HashMap<>();
```

Comment:
I think it would make sense to split this between methods and fields, using 2 maps instead of 1.